hal/src/electron/modem/mdm_hal.cpp

```diff
@@ -950,7 +951,7 @@ bool MDMParser::init(DevStatus* status)
         goto failure;
     // Request IMSI (International Mobile Subscriber Identification)
     sendFormated("AT+CIMI\r\n");
-    if (RESP_OK != waitFinalResp(_cbString, &str_imsi))
+    if (RESP_OK != waitFinalResp(_cbString, &str_imsi, CIMI_TIMEOUT))
```

Comment:
For consistency, might need to add `CIMI_TIMEOUT` here?

https://github.com/particle-iot/device-os/blob/3b3984ffe000130ba7c64b7811d76a1be688d84e/hal/src/electron/modem/mdm_hal.cpp#L1529

https://github.com/particle-iot/device-os/blob/3b3984ffe000130ba7c64b7811d76a1be688d84e/hal/src/electron/modem/mdm_hal.cpp#L1474

Reply:
Sounds good.  I found another place to add it as well.  Should we have the callback added here?
```c
                // Query IMSI every 60 sec if not regsitered, to detect IMSI switches
                if (HAL_Timer_Get_Milli_Seconds() - start_imsi_check > 60000UL) {
                    start_imsi_check = HAL_Timer_Get_Milli_Seconds();
                    sendFormated("AT+CIMI\r\n");
                    if (RESP_OK != waitFinalResp()) {
                        goto failure;
                    }
                }
```

Reply:
I don't think that it is needed to add a callback here because it's only for debugging, and we will not use this IMSI for other purposes. But once the device registers on a particular IMSI, we would want that IMSI stored because that's the IMSI that has worked for registration, and in that part of the code, there is the callback. (https://github.com/particle-iot/device-os/blob/23c2920ac3263363a00ab0595db6a17eaf0c2964/hal/src/electron/modem/mdm_hal.cpp#L1529)

Reply:
Ok sounds good, I'll just add the explicit timeout `waitFinalResp(nullptr, nullptr, CIMI_TIMEOUT);` which is the same as the default timeout, but better to be explicit with the issue with CIMI.

Reply:
👍  perfecto